Fix issues when building with system Python 3

ghweber opened the pull request:
### Description

Resolve various issues when trying to build with system Python 3.x:

   * Use `python3` and `python3-config`
   * Use `--embed` option when calling `python3-config` (otherwise Python library is omitted for `--libs` and the library name detection fails.
   * Use `which` to get full path of `PYTHON_COMMAND` and `PYTHON_CONFIG_COMMAND` (at lease I ran into some problems when `PYTHON_COMMAND` was not a fully qualified path.
   * Do not require `openssl` and `zlib` as dependencies when using system Python.
  
### Type of change

Bug fix

### How Has This Been Tested?

Please describe the tests you've added or any tests that already cover this change. Include relevant information, such as which operating system you tested on.

### Checklist:

- [x] I have followed the [style guidelines][1] of this project.~~
- [x] I have performed a self-review of my own code.~~
- [ ] I have commented my code where applicable.~~
- [ ] I have updated the release notes.~~
- [ ] I have made corresponding changes to the documentation.~~
- [ ] I have added debugging support to my changes.~~
- [ ] I have added tests that prove my fix is effective or that my feature works.~~
- [ ] I have confirmed new and existing unit tests pass locally with my changes.~~
- [ ] I have added any new baselines to the repo.~~
- [x] I have NOT made any changes to [*protocol* or *public interfaces*][3] in an RC branch.~~
- [ ] I have assigned reviewers (see [VisIt's PR procedures][2] for more information).~~

[1]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/StyleGuide.html
[2]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/pr_create.html#reviewers
[3]: https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/dev_manual/RCDevelopment.html#communication-protocols-and-public-apis

ghweber committed:
Add required --embed option when calling python3-config for Python versions >=3.8

biagas commented on src/tools/dev/scripts/bv_support/bv_python.sh:
```diff
@@ -80,7 +80,11 @@ function python_set_vars_helper
+    if [[ "$DO_PYTHON2" == "yes" ]] ; then
+        PYTHON_LIBRARY=`"$PYTHON_CONFIG_COMMAND" --libs`
+    else
+        PYTHON_LIBRARY=`"$PYTHON_CONFIG_COMMAND" --libs --embed`
```
--embed doesn't work on all systems (just tried on our LC system).
And it causes the 'usage' text to appear and the command to fail.

ghweber replied:
Even for Python 3? Then to be honest, I do not know what to change. On my system (and probably at NERSC), the `--embed` option is necessary as otherwise `-lpython.x.y` link option is **not** included and the whole build fails because the subsequent attempt to extract the Python library name fails. I am wondering if they added it for a later Python 3.x version. What version is installed on LC?

ghweber replied:
Some googling, it seems to have changed with Python 3.8: https://bugs.python.org/issue36721, so I guess we should add the option based on the exact Python version and not just Python 2 vs. Python 3.

ghweber commented:
Hi Kathleen @biagas, I noticed that I accidentally erased the line
```
    echo $PYTHON_BUILD_DIR $PYTHON_VERSION $VISIT_PYTHON_DIR
```
in `python_set_vars_helper` since I thought I had added it as debug output. However, it seems that it is actually propagating information to the caller. I added a commit that reverts that deletion. Can you please review again?

biagas commented:
Good catch!

markcmiller86 commented:
@ghweber can you please take a look at conflicts here and resolve them? If you can, I can try to get it merged.

ghweber commented:
@markcmiller86 not sure why this didn't get merged in May after Kathleen's approval. May have been because I think she would do it and that I wasn't sure if I am even permitted to do merges. Anyway, the conflicts are there because she seems to have fixed most things independently. The only issue that remains is that for Python 3.8 and newer there is the need to add the `--embed` option when calling `python3-config`. Otherwise it will miss the actual Python library. I updated the merge request to do only that.

biagas committed:
Merge branch 'develop' into develop
